velox/exec/tests/RowContainerTest.cpp

```diff
@@ -47,29 +74,102 @@ class RowContainerTest : public exec::test::RowContainerTestBase {
 
     auto result = BaseVector::create(expected->type(), size, pool_.get());
     container.extractColumn(oddRows.data(), size, column, result);
-    EXPECT_EQ(size, result->size());
-    for (vector_size_t row = 0; row < size; ++row) {
-      if (row % 2 == 0) {
-        EXPECT_TRUE(result->isNullAt(row)) << "at " << row;
+    return result;
+  }
+
+  VectorPtr testExtractColumnRowNumbersForOddRows(
+      RowContainer& container,
+      const std::vector<char*>& rows,
+      int column,
+      const VectorPtr& expected) {
+    auto size = rows.size();
+
+    // Set the first row in 'oddRows' to be null, and all the
+    // even row numbers point to it.
+    std::vector<char*> oddRows(size, nullptr);
+    for (auto i = 1; i < size; i++) {
+      oddRows[i] = rows[i];
+    }
+
+    // Extract only odd rows.
+    // Test using extractColumn (with rowNumbers) API.
+    // The rowNumbersBuffer has values like 0, 1, 0, 3, 0, 5, 0, 7, etc
+    // This tests the case that the rowNumber values are repeated and are
```

Comment:
Do you have a use case for extracting duplicate row numbers?

Reply:
Yes, we should support extracting duplicate row numbers if required. So I constructed this case in this manner.